tahrir_api/scripts/populatedb.py

```diff
@@ -0,0 +1,217 @@
+
+import datetime
+import os
+import random
+import sys
+import transaction
+
+from sqlalchemy import engine_from_config
+from sqlalchemy.orm import scoped_session, sessionmaker
+from paste.deploy import appconfig
+from zope.sqlalchemy import ZopeTransactionExtension
+
+from ..model import (
+    Issuer,
+    Badge,
+    Person,
+    Team,
+    Milestone,
+    Series,
+    Assertion,
+    DeclarativeBase,
+)
+from ..utils import convert_name_to_id
+
+
+DBSession = scoped_session(sessionmaker(extension=ZopeTransactionExtension(),
+                                        expire_on_commit=False))
+
+
+def usage(argv):
+    cmd = os.path.basename(argv[0])
+    print('usage: %s <config_uri>\n'
+          '(example: "%s development.ini")' % (cmd, cmd))
+    sys.exit(1)
+
+
+def _getpathsec(config_uri, name):
+    if '#' in config_uri:
+        path, section = config_uri.split('#', 1)
+    else:
+        path, section = config_uri, 'main'
+    if name:
+        section = name
+    return path, section
+
+
+def main(argv=sys.argv):
```

Comment:
This function is extremely long and difficult to follow as a result. I recommend breaking it up into small helper functions that each do a small logical block. Then each function will be simple and easy to understand (and much easier to test!)